PYIC 4188: changed to the cimit-stub lambdas for return codes

TalatAydinGDS opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed

changed the PutContraIndicator lambda for storage the issuers in DynamoDB

### Why did it change

So, can add issuers within the response from GetContraIndicatorCredential CimitStub's lambda, similar to how it is done in the SPOT CIMIT's response

### Issue tracking
<!-- List any related Jira tickets or GitHub issues -->
<!-- List any related ADRs or RFCs -->
<!-- Delete/copy as appropriate -->

- [PYIC-4188](https://govukverify.atlassian.net/browse/PYIC-4188)

## Checklists

### Environment variables or secrets

<!-- Delete if changes DO include new environment variables or secrets -->
- [ ] No environment variables or secrets were added or changed

<!-- Delete if changes DO NOT include new environment variables or secrets -->
- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks


[PYIC-4188]: https://govukverify.atlassian.net/browse/PYIC-4188?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

TalatAydinGDS committed:
PYIC 4188: changed to the cimit-stub lambdas for return codes

Joe-Edwards-GDS commented on di-ipv-cimit-stub/lambdas/put-contra-indicators/src/main/java/uk/gov/di/ipv/core/putcontraindicators/service/ContraIndicatorsService.java:
```diff
@@ -128,15 +124,19 @@ private ContraIndicatorEvidenceDto parseContraIndicatorEvidence(SignedJWT signed
+            throws ParseException {
 
+        Instant issuanceDate = getIssuanceDate(signedJWT.getJWTClaimsSet().getNotBeforeTime());
+        String iss = signedJWT.getJWTClaimsSet().getClaim(ISSUER).toString();
```
Can we use `getIssuer()` on the claims set?

TalatAydinGDS replied:
yeah, thanks

Joe-Edwards-GDS commented on di-ipv-cimit-stub/lambdas/put-contra-indicators/src/test/java/uk/gov/di/ipv/core/putcontraindicators/service/ContraIndicatorsServiceTest.java:
```diff
@@ -63,19 +67,20 @@ public void addUserCisShouldInsert() {
                         .signedJwt(SIGNED_CRI_VC)
                         .build();
-
+        String iss = SignedJWT.parse(SIGNED_CRI_VC).getJWTClaimsSet().getClaim(ISSUER).toString();
```
ditto

Joe-Edwards-GDS commented on di-ipv-cimit-stub/lambdas/put-contra-indicators/src/main/java/uk/gov/di/ipv/core/putcontraindicators/service/ContraIndicatorsService.java:
I think we still need to update the `saveOrUpdateCimitStubItems` method, so that it either sets or updates the issuer list?

TalatAydinGDS replied:
Yes, definitely :) Thank you

Joe-Edwards-GDS reviewed (commented):
A few more comments but getting there I think

TalatAydinGDS committed:
PYIC 4188: code refactor according code reviews

Wynndow commented on di-ipv-cimit-stub/lambdas/stub-management/src/main/java/uk/gov/di/ipv/core/stubmanagement/service/UserService.java:
```diff
@@ -38,11 +38,12 @@ public void addUserCis(String userId, List<UserCisRequest> userCisRequest) {
                         cimitStubItemService.persistCimitStub(
                                 userId,
                                 user.getCode().toUpperCase(),
+                                null,
```
Should this be null? This is called when a CI is added with the management API. I think the user who is submitting the request will need to set the issuers on the CI they want to create, and then we pass that through here to be stored.

Wynndow replied:
I think we'll also need something in the `else` branch here to set allow the issuer to be updated?

Wynndow commented on di-ipv-cimit-stub/lambdas/stub-management/src/main/java/uk/gov/di/ipv/core/stubmanagement/service/UserService.java:
```diff
@@ -72,6 +73,7 @@ public void updateUserCis(String userId, List<UserCisRequest> userCisRequest) {
                     cimitStubItemService.persistCimitStub(
                             userId,
                             user.getCode().toUpperCase(),
+                            null,
```
Again - I think issuers needs to be stored from the request here.

Wynndow reviewed (commented):
Just a few comments about handling requests in the management API

Wynndow committed:
PYIC-4188: Allow management API to set issuers

The CiMit stub will now be returning issuers in its returned VC.

This allows values for the issuers to be set with the management API.

Wynndow commented:
@TalatAydinGDS Are you able to deploy the cimit stub to your dev env? I'm struggling with mine. Would be good to get this deployed and check it does what we think it does before merging.

TalatAydinGDS commented:
@Wynndow I deployed to my dev environment and test it. I checked in DynamoDB and logs. Additionally, We did a call with @SreeBandla01 and ran feature tests with him from my dev environment and checked together to log and DynamoDB in my dev environment.

Wynndow commented:
> @Wynndow I deployed to my dev environment and test it. I checked in DynamoDB and logs. Additionally, We did a call with @SreeBandla01 and ran feature tests with him my dev environment and checked together to log and DynamoDB in my dev environment.

OK great. I want to just check the new functionality I added in my commit. I've just managed to deploy it so will do that then approve.
